src/dependency_specifications.rb

```diff
@@ -0,0 +1,31 @@
+
+# frozen_string_literal: true
+
+module Metalware
+  # Class to generate reused `dependency_hash`s, for use by `Dependency` object
+  # to enforce command dependencies.
+  # XXX Consider moving generating of more `dependency_hash`s here.
+  class DependencySpecifications
+    def initialize(config)
+      @config = config
+    end
+
+    def for_node_in_configured_group(node_name)
+      primary_group = configured_primary_group_for_node(node_name)
+      {
+        repo: ['configure.yaml'],
+        configure: ['domain.yaml', "groups/#{primary_group}.yaml"],
```

Comment:
Looks like that branch is missing the following commit. Should work otherwise.
https://github.com/alces-software/metalware/pull/138/commits/717207279fde9dad94725304eec7e410b3ade831